jsonschema-core.xml

```diff
@@ -363,15 +398,26 @@
                     This value SHOULD be normalized, and SHOULD NOT be an empty fragment &lt;#&gt; or an empty string &lt;&gt;.
                 </t>
                 <t>
-                    The root schema of a JSON Schema document SHOULD contain an "$id" keyword with an absolute-URI (containing a scheme, but no fragment).
+                    The root schema of a JSON Schema document SHOULD contain an "$id" keyword with 
+                    a URI (containing a scheme).  This URI SHOULD either not have a fragment, or
+                    have one that is an empty string.
+                    <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
+                    <cref>
+                        How should an "$id" URI reference containing a fragement with other components
+                        be interpreted?  There are two cases:  when the other components match
+                        the current base URI and when they change the base URI.
+                    </cref>
                 </t>
                 <t>
                     To name subschemas in a JSON Schema document,
                     subschemas can use "$id" to give themselves a document-local identifier.
-                    This form of "$id" keyword MUST begin with a hash ("#") to identify it as a fragment URI reference,
-                    followed by a letter ([A-Za-z]), followed by any number of
-                    letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), or periods (".").
-                    <!-- This restriction is the same one defined by XML -->
+                    This is done by setting "$id" to a URI reference consisting only of a fragment.
+                    The fragment identifier MUST NOT match the syntax of a JSON Pointer fragment
+                    identifier.
+                </t>
+                <t>
+                    The effect of defining an "$id" that is appears to be a JSON Pointer that does not match
+                    the expected JSON Pointer for that schema is not defined.
```

Comment:
@awwright I'm happy to restore the XML-derived restrictions on plain name fragments.  The possibility of JSON defining something different is a really good point.